### ISSUE TITLE
The AML extension can be loaded as a RemoteAuthorityResolver for the "vscode-remote://amlext" scheme/authority

### ISSUE THREAD

**Issue by tbombach** [microsoft/vscode-tools-for-ai#742]
The AML extension needs to contribute a `RemoteAuthorityResolver` that VSCode uses to start a remote session. This resolver will point to the local proxy, which will point to the compute instance proxy.

- Follow the example from the codespaces extension, which implements the `RemoteAuthorityResolver` as `CloudEnvResolver`
  - [x] We need to register this resolver, along with the authority that we intend to process. For us, it will be `amlext`. This lets vscode search for an extension that knows how to handle urls like `vscode-remote://amlext`
  - [x] We need to implement the resolver so that if it gets a url like `vscode-remote://amlext`, it tells vscode to connect to a specific web address. In our case, it will be the proxy at localhost:3003 (or whatever port is open)

**Comment by tbombach** [comment]
**Roadblock**: VS Code is not loading the AML extension when we open a new window via the `openFolder` command. It says that there is no extension registered for the authority `amlext`. This should not be happening because we call `registerRemoteAuthorityResolver` and pass along the authority `amlext`

**Comment by tbombach** [comment]
Figured this out! The AML extension has dependencies on the azure-account and python extension in package.json. However, calling `RemoteAuthorityResolver` is one of the first things that happens when trying to resolve a vscode-remote:// url (because extensions can be on the remote server, so vscode needs to determine where to connect to first). I think azure-account and python extensions don't have the extensionKind value in package.json (e.g. ui or workspace), so I'm not sure if they're loaded in the way we want for the remote scenario.  I have removed the dependencies from our extension for now, and vscode now loads our extension and calls our remote authority resolver!
